osu.Game/Screens/Play/Player.cs

```diff
@@ -285,6 +287,10 @@ private void load(AudioManager audio, OsuConfigManager config, OsuGameBase game)
             ScoreProcessor.HasCompleted.ValueChanged += updateCompletionState;
             HealthProcessor.Failed += onFail;
 
+            // Keep track of whether the storyboard ended after the playable portion
+            if (DimmableStoryboard.HasStoryboardEnded != null)
+                DimmableStoryboard.HasStoryboardEnded.ValueChanged += updateCompletionState;
```

Comment:
I think it would suffice to have a separate `HasStoryboardEnded` bindable in `DimmableStoryboard`, defaulting to `true`, and just bind to it normally on storyboard load, instead of directly proxying as done currently. Would also spare the null check here, too.

Reply:
One thing is that I wasn't sure about letting `DimmableStoryboard` expose bindables from `DrawableStoryboard` by creating each and bind on storyboard creation, but instead leave each class do its own thing.

I guess your solution is fine for the fact that `DimmableStoryboard` is still a storyboard-related class.